tests/test_pyfunction.rs

```diff
@@ -558,3 +558,30 @@ fn test_reference_to_bound_arguments() {
         py_assert!(py, function, "function(1, 2) == 3");
     })
 }
+
+#[test]
+fn test_pyfunction_raw_ident() {
+    #[pyfunction]
+    fn r#struct() -> bool {
+        true
+    }
+
+    #[pyfunction]
+    #[pyo3(name = "r#enum")]
```

Comment:
Personally it feels a bit weird to me that this quoted string accepts the `r#` escape. Maybe the fix in #4226 is preferable?

Reply:
I also feel like #4226 is the better solution. But I was also under the impression that they are not mutually exclusive? So we could support raw identifiers as proposed here but teach #4226 as the cleaner API surface?

Reply:
I'm definitely happy to have both if that feels desirable. This syntax being accepted here is surprising to me but I don't think unreasonable.

Reply:
Yes, this was not meant to replace #4226. It was just meant to fix the inconsistency to `#[pyo3(name = ...)]` on `#[pymethods]`, which already works this way. I generally would expect the `name` attribute to allow anything that the function name does also allow. Also, this syntax does compile currently, but does not give a usable function on the Python side, which I think is just bad. I'm also fine if we want to explicitly reject this, but it seems easier to just make it work than to implement the error handling.